core/templates/self_list.h

```diff
@@ -105,6 +105,57 @@ class SelfList {
 			}
 		}
 
+		void sort() {
+			sort_custom<Comparator<T>>();
+		}
+
+		template <class C>
+		void sort_custom() {
+			if (_first == _last) {
+				return;
+			}
+
+			SelfList<T> *from = _first;
+			SelfList<T> *current = from;
+			SelfList<T> *to = from;
+
+			while (current) {
+				SelfList<T> *next = current->_next;
+
+				if (from != current) {
+					current->_prev = nullptr;
+					current->_next = from;
+
+					SelfList<T> *find = from;
+					C less;
+					while (find && less(*(find->_self), *(current->_self))) {
```

Comment:
I don't know, I find it clearer the way it is right now. But not sure what we usually do it that case.

Reply:
Yeah I think explicit is better in this case. If you're not very clear on the precedence order, this can easily backfire.

Reply:
There are no instances of this syntax anywhere I can find in the engine at least, but many instances of the suggested syntax, for example in `array.cpp`

Reply:
> There are no instances of this syntax anywhere I can find in the engine at least, but many instances of the suggested syntax, for example in array.cpp

Yeah, I checked too and I could not find anywhere where the explicit parenthesis are used. I can update for consistency I guess.

Reply:
Also there's only really confusion possible here if `find` or `current` is a pointer to a pointer/reference as otherwise you wouldn't have `->` operator